### PR TITLE
Server-client discovery use-case [7789]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/177/)

File: code/CodeTester.cpp [hunk 1]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
+Locator_t server_locator;
+IPLocator::setIPv4(server_locator, "192.168.10.57");
+server_locator.port = 56542;
+
+RemoteServerAttributes server_attr;
+server_attr.ReadguidPrefix("72.61.73.70.66.61.72.6d.74.65.73.74");
+server_attr.metatrafficUnicastLocatorList.push_back(server_locator);
+
+participant_attr.rtps.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::SERVER;
+participant_attr.rtps.builtin.discovery_config.m_DiscoveryServers.push_back(server_attr);
```
Comment (EduPonz) [inline code/CodeTester.cpp]: This is not needed nor correct. You'd do something like this to connect one server to another
```suggestion
```

File: code/CodeTester.cpp [hunk 2]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
-//CONF_DISCOVERY_SERVER
+//CONF_DISCOVERY_SERVER_MAIN_SCENARIO
+
+//CONF_DS_MAIN_SCENARIO_SERVER
+Locator_t server_locator;
+IPLocator::setIPv4(server_locator, "192.168.10.57");
+server_locator.port = 56542;
+
+RemoteServerAttributes server_attr;
+server_attr.ReadguidPrefix("72.61.73.70.66.61.72.6d.74.65.73.74");
```
Comment (EduPonz) [inline code/CodeTester.cpp]:
```suggestion
participant_attr.rtps.ReadguidPrefix("72.61.73.70.66.61.72.6d.74.65.73.74");
```

File: code/CodeTester.cpp [hunk 3]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
+server_attr.metatrafficUnicastLocatorList.push_back(server_locator);
```
Comment (EduPonz) [inline code/CodeTester.cpp]:
```suggestion
participant_attr.rtps.metatrafficUnicastLocatorList.push_back(server_locator);
```

File: code/CodeTester.cpp [hunk 4]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
 
-//CONF_DISCOVERY_SERVER
+//CONF_DISCOVERY_SERVER_MAIN_SCENARIO
+
+//CONF_DS_MAIN_SCENARIO_SERVER
+Locator_t server_locator;
+IPLocator::setIPv4(server_locator, "192.168.10.57");
+server_locator.port = 56542;
+
+RemoteServerAttributes server_attr;
```
Comment (EduPonz) [inline code/CodeTester.cpp]:
```suggestion
```

File: code/CodeTester.cpp [hunk 5]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
+server_attr.ReadguidPrefix("72.61.73.70.66.61.72.6d.74.65.73.74");
+server_attr.metatrafficUnicastLocatorList.push_back(server_locator);
+
+participant_attr.rtps.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::SERVER;
+participant_attr.rtps.builtin.discovery_config.m_DiscoveryServers.push_back(server_attr);
+participant_attr.rtps.builtin.discovery_config.discoveryServer_client_syncperiod = Duration_t(0,250000000);
```
Comment (EduPonz) [inline code/CodeTester.cpp]:
```suggestion
participant_attr.rtps.builtin.discovery_config.discoveryServer_client_syncperiod = Duration_t(0, 250000000);
```

File: code/CodeTester.cpp [hunk 6]
```diff
@@ -456,7 +456,39 @@ meta_unicast_locator.port = 7412;
...
+IPLocator::setIPv4(server_locator, "192.168.10.57");
+server_locator.port = 56542;
+
+RemoteServerAttributes remote_server_attr;
+remote_server_attr.ReadguidPrefix("72.61.73.70.66.61.72.6d.74.65.73.74");
+remote_server_attr.metatrafficUnicastLocatorList.push_back(server_locator);
+
+participant_attr.rtps.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::CLIENT;
+participant_attr.rtps.builtin.discovery_config.m_DiscoveryServers.push_back(remote_server_attr);
+participant_attr.rtps.builtin.discovery_config.discoveryServer_client_syncperiod = Duration_t(0,250000000);
```
Comment (EduPonz) [inline code/CodeTester.cpp]:
```suggestion
participant_attr.rtps.builtin.discovery_config.discoveryServer_client_syncperiod = Duration_t(0, 250000000);
```

File: code/XMLTester.xml [hunk 7]
```diff
@@ -1457,10 +1457,64 @@
 
 <!--><-->
 
-<!-->CONF_DISCOVERY_SERVER<-->
+<!-->CONF_DISCOVERY_SERVER_MAIN_SCENARIO<-->
 
+<!-->CONF_DS_MAIN_SCENARIO_SERVER<-->
```
Comment (EduPonz) [inline code/XMLTester.xml]: The `discoveryServer_client_syncperiod` is specified in the C++, but not in the XML

File: docs/use-cases.rst [hunk 8]
```diff
@@ -135,6 +134,56 @@ the DDS standard to adapt it to wide deployment scenarios.
...
 
+.. START SUBSEC:DISCOVERY-SERVER
+
+.. _server-client-discovery-use-case:
+
+Server-Client Discovery
+-----------------------
+
+Considering a scenario in which a large number of communication agents, called participants in this case, are deployed,
+an alternative to the default SIMPLE discovery mechanism of the RTPS standard may be used. For this purpose, Fast RTPS
```
Comment (EduPonz) [inline docs/use-cases.rst]:
```suggestion
an alternative to the default RTPS standard SIMPLE discovery mechanism may be used. For this purpose, Fast-RTPS
```

File: docs/use-cases.rst [hunk 9]
```diff
@@ -135,6 +134,56 @@ the DDS standard to adapt it to wide deployment scenarios.
...
+provides for the client-server discovery mechanism, in which a server participant operates as the central point of
```
Comment (EduPonz) [inline docs/use-cases.rst]:
```suggestion
provides a client-server discovery mechanism, in which a server participant operates as the central point of
```

File: docs/use-cases.rst [hunk 10]
```diff
@@ -135,6 +134,56 @@ the DDS standard to adapt it to wide deployment scenarios.
...
+communication; the server collects and processes the metatraffic sent by the client participants and used in the
+discovery phases, and distributes the appropriate information to the previous participants.
```
Comment (EduPonz) [inline docs/use-cases.rst]:
```suggestion
communication; the server collects and processes the metatraffic sent by the client participants, and distributes the appropriate information to the previous participants.
```

File: docs/use-cases.rst [hunk 11]
```diff
@@ -135,6 +134,56 @@ the DDS standard to adapt it to wide deployment scenarios.
...
+
+Various discovery server use cases are presented below.
+
+.. _discovery_server_major_scenario_setup:
+
+Discovery-Server major scenario setup
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (EduPonz) [inline docs/use-cases.rst]: I think this is a UDPv4 example. I'd call it something on those lines

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/179/)

File: docs/use-cases.rst [hunk 12]
```diff
@@ -135,6 +134,56 @@ the DDS standard to adapt it to wide deployment scenarios.
...
+.. _server-client-discovery-use-case:
+
+Server-Client Discovery
+-----------------------
+
+Considering a scenario in which a large number of communication agents, called participants in this case, are deployed,
+an alternative to the default RTPS standard SIMPLE discovery mechanism may be used. For this purpose, Fast-RTPS
+provides a client-server discovery mechanism, in which a server participant operates as the central point of
+communication; the server collects and processes the metatraffic sent by the client participants, and distributes the
+appropriate information to the previous participants.
```
Comment (EduPonz) [inline docs/use-cases.rst]:
```suggestion
communication, that is the server collects and processes the metatraffic sent by the client participants, and distributes the
appropriate information among the rest of the clients.
```

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/181/)

**Pull request merged**